File: submissions/217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> numbers = new HashSet<Integer>();
        Boolean isValid = false;
        for (int i = 0; i < nums.length; i++) {
            if (numbers.contains(nums[i])) {
                isValid = true;
            } else {
                numbers.add(nums[i]);
            }
        }
        return isValid;
    }
}

File: submissions/217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> seen;
        const int n = nums.size();
        seen.reserve(n);

        for(int x : nums) {
            if(!seen.insert(x).second) return true;
        }
        return false;
    }
};
